wagtail/wagtailcore/models.py

```diff
@@ -58,9 +58,30 @@ def get_by_natural_key(self, hostname, port):
 @python_2_unicode_compatible
 class Site(models.Model):
     hostname = models.CharField(verbose_name=_('Hostname'), max_length=255, db_index=True)
-    port = models.IntegerField(verbose_name=_('Port'), default=80, help_text=_("Set this to something other than 80 if you need a specific port number to appear in URLs (e.g. development on port 8000). Does not affect request handling (so port forwarding still works)."))
+    port = models.IntegerField(
+        verbose_name=_('Port'),
+        default=80,
+        help_text=_(
+            "Set this to something other than 80 if you need a specific port number to appear in URLs (e.g. "
+            "development on port 8000). Does not affect request handling (so port forwarding still works)."
+        )
+    )
+    site_name = models.CharField(
+        verbose_name=_('Site name'),
+        max_length=255,
+        db_index=True,
```

Comment:
Will leave this db_index out, as I don't think it's likely that code will regularly be doing lookups by human-readable name. (Our title fields elsewhere don't have an index.)